### ISSUE TITLE
Report sometimes includes irrelevant posts

### ISSUE THREAD

**Issue by iceaway** [mathisdt/trackworktime#198]
Hi!

I have noticed that sometimes when doing a report for the last (or current) week, it will include days of other weeks to. I just made an report of the "current" week, and I got this data included in my CSV-file:

```
2022-02-28 <--- OK
2022-03-01 <--- OK
-12-21     <--- corrupted
2021-12-22 <--- should not be included
2021-12-16 <--- should not be included
2021-12-17 <--- should not be included
```

Occasionally I also get a single number on the last line.

**Comment by mathisdt** [comment]
Hi @iceaway, thanks for the report! I tried reproducing this particular problem but was unsuccessful. Could you please verify that you use the latest version of the app ([1.1.12](https://github.com/mathisdt/trackworktime/releases/tag/v1.1.12)), then reproduce the error and send me your log file? Maybe I can use some information from it to get a clue what's going wrong here. Thanks!

**Comment by iceaway** [comment]
I just sent you an e-mail with the logfile. I checked my TWT and it is indeed the latest version, 1.1.12.

**Comment by mathisdt** [comment]
Thanks for file log! Strangely enough I can't find anything unusual in the log file. Do you use Magisk or any other "non-standard" software on your phone? I'm afraid that I can't help you as I don't know the problem...

**Comment by mathisdt** [comment]
Another question: do you see this currupted data only inside the CSV file or also in the preview sometimes?

**Comment by iceaway** [comment]
I'm not using Magisk,. I'm not 100% sure about the preview unfortunately, but I think I have seen it there as well. I will make sure I check next time. I understand that nothing can be done if you see nothing wrong in the logs / cannot reproduce. Thanks for taking the time! Maybe it will resolve itself in the future (no huge deal anyway, I can just remove the bad rows from the export).

**Comment by iceaway** [comment]
Hi again! I just got this error again and I checked the Preview vs. the exported file, and the preview was fine but the exported .csv-file was corrupted. Not sure if this information helps!

**Comment by mathisdt** [comment]
It helps, because it determines that the problem is only in the CSV *file export* and not in the generation of the data. If generating the data would have the problem, then also the preview would contain the errors.

Could it be that your data sometimes contains non-ASCII characters? I found out that the app uses the system encoding which may have problems with these characters (I'll push a fix for that anyway, but it would be interesting if there's a connection).

**Comment by iceaway** [comment]
I have one task that has a non-ascii letter in it (ö), but it was not used in the time period that I exported. I also tried to rename that task with an 'o' instead of an 'ö', but when I did the export the .csv-file was once again corrupted. It's interesting that it also brings in dates from other time periods (i.e. when I generate "last week" I get a line with an entry from 2022-02-18 among others).

**Comment by mathisdt** [comment]
Debug information: the error occurs only in the files written by the app, not in the preview!

**Comment by DocOellerson** [comment]
Hi there, 
I have the exact same problem with generating reports. Preview is fine, the csv file contains stuff from the last report - seems like something isn't flushed correctly and the older report was longer than the newer one. 
Version is 1.1.14

See this part of my work time report from april:
[...]
2022-04-27T15:00:20+02:00;out;;
2022-04-28T07:36:20+02:00;in;Arbeit;
2022-04-28T15:38:25+02:00;out;;
2022-04-29T07:34:45+02:00;in;Arbeit;
2022-04-29T07:35:42+02:00;out;;
2022-04-29T07:39:45+02:00;in;Arbeit;
2022-04-30T08:59:42+02:00;out;;
eit;
2022-03-22T15:57:11+01:00;out;;
2022-03-23T12:21:27+01:00;in;Arbeit;
2022-03-23T12:54:18+01:00;out;;
2022-03-23T12:55:47+01:00;in;Arbeit;
2022-03-23T12:56:47+01:00;out;;
2022-03-24T10:47:05+01:00;in;Arbeit;
2022-03-24T15:30:07+01:00;out;;

It looks like the last 8 lines are the remains of an older report from march. The line containing "eit;" seems to be the tail of "Arbeit;".
Regards, 
Docoellerson

**Comment by mathisdt** [comment]
I think I found the problem, although I can't be sure because I can't reproduce this. If the bug is also in version 1.1.5 which should be published soon(ish) then please reopen this issue. Thanks for the detailed bug reports! @iceaway @DocOellerson

**Comment by iceaway** [comment]
Great work @mathisdt, will report back as soon as I have tested the new version.

**Comment by mathisdt** [comment]
Now I finally could reproduce this:
1. generate any report
2. generate the *exact same report* again (same options selected in report creation dialog), but the output has to be smaller this time (in the "events: raw" report, this can be archived by deleting some events or removing a comment which was previously there)

**Result:** the last characters from the first report are still there in the second. The file size stays the same, only the beginning of the report is different.

**Explanation:** Internally the report is written to the same file (because the file name is generated from the parameters selected for the report). This file was opened in a wrong mode ("write" instead of "write-truncate"). This is fixed with the above commit, and the reports only contain the intended contents. The fix will be released with version 1.1.15.
